### PR TITLE
[15.0][ADD] purchase_order_line_menu: New views and stored fields

### PR DESCRIPTION
**Pull request by Shide**
New views and fields added to the purchase order line menu

MT-2028 @moduon @rafaelbn @yajo @dreispt please review and comment if you want  :)

### CODE REVIEW

**Comment by rafaelbn** [comment]
Ping @max3903 🤗 we will improve also same for sales

**Comment by Shide** [comment]
Related to https://github.com/OCA/sale-workflow/pull/2591

File: purchase_order_line_menu/models/purchase_order_line.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    _inherit = "purchase.order.line"
+
+    invoice_status = fields.Selection(
+        related="order_id.invoice_status",
+        readonly=True,
+        store=True,
+    )
+    date_order = fields.Datetime(
+        store=True,
+    )
```
Comment (yajo) [inline purchase_order_line_menu/models/purchase_order_line.py]:
```suggestion
    invoice_status = fields.Selection(
        related="order_id.invoice_status",
        index=True,
        readonly=True,
        store=True,
    )
    date_order = fields.Datetime(
        index=True,
        store=True,
    )
```

File: purchase_order_line_menu/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -1 +1 @@
-Adds a menu item to list all Purchase Order lines.
+Adds a menu item and a couple of views to navigate through Purchase Order lines.
```
Comment (yajo) [inline purchase_order_line_menu/readme/DESCRIPTION.rst]:
```suggestion
Adds a menu item and some views to navigate through Purchase Order lines.
```

File: purchase_order_line_menu/views/purchase_order_line_views.xml [hunk 3]
```diff
@@ -1,20 +1,178 @@
...
+                    string="Planned date"
+                    date="date_planned"
+                />
+                <separator />
+            </xpath>
+            <xpath expr="//group" position="inside">
+                <filter
+                    name="groupby_invoice_status"
+                    string="Billing Status"
+                    domain="[]"
```
Comment (yajo) [inline purchase_order_line_menu/views/purchase_order_line_views.xml]:
```suggestion
```

File: purchase_order_line_menu/views/purchase_order_line_views.xml [hunk 4]
```diff
@@ -1,20 +1,178 @@
...
+                    context="{'group_by' : 'invoice_status'}"
+                />
+                <filter
+                    name="groupby_account_analytic"
+                    string="Analytic Account"
+                    domain="[]"
```
Comment (yajo) [inline purchase_order_line_menu/views/purchase_order_line_views.xml]:
```suggestion
```

File: purchase_order_line_menu/views/purchase_order_line_views.xml [hunk 5]
```diff
@@ -1,20 +1,178 @@
...
+                    context="{'group_by' : 'account_analytic_id'}"
+                />
+                <filter
+                    name="groupby_analytic_tags"
+                    string="Analytic Tags"
+                    domain="[]"
```
Comment (yajo) [inline purchase_order_line_menu/views/purchase_order_line_views.xml]:
```suggestion
```

File: purchase_order_line_menu/views/purchase_order_line_views.xml [hunk 6]
```diff
@@ -1,20 +1,178 @@
...
+                    context="{'group_by' : 'analytic_tag_ids'}"
+                />
+                <filter
+                    name="groupby_date_order"
+                    string="Order Date"
+                    domain="[]"
```
Comment (yajo) [inline purchase_order_line_menu/views/purchase_order_line_views.xml]:
```suggestion
```

**Review by yajo — APPROVED** [review]
Some minor suggestions. Code review.

**Review by rafaelbn — APPROVED** [review]
👍🏼 Working in production

Please @dreispt @yajo could you update your review and merge?

Similar to the already merged for sales https://github.com/OCA/sale-workflow/pull/2591

**Review by dreispt — APPROVED** [review]
Nitpick: should be an IMP rather than an ADD. 
ADD would be a new module, IMP is adding a new feature to an existing module ;)

**Pull request merged**